sdk/servicebus/azure-messaging-servicebus/src/test/java/com/azure/messaging/servicebus/ServiceBusReceiverAsyncClientTest.java

```diff
@@ -122,6 +131,41 @@ void teardown() {
         consumer.close();
     }
 
+    /**
+     * Verifies that this peek one messages.
+     */
+    @Test
+    void peekOneMessage() {
+        // Arrange
+        final int numberOfEvents = 1;
+        when(managementNode.peek())
+            .thenReturn(Mono.just(mock(ServiceBusReceivedMessage.class)));
+
+        // Act & Assert
+        StepVerifier.create(consumer.peek())
+            .expectNextCount(numberOfEvents)
+            .verifyComplete();
+
+    }
+
+    /**
+     * Verifies that this peek one messages from a sequence Number.
+     */
+    @Test
+    void peekWithSequenceOneMessage() {
+        // Arrange
+        final int numberOfEvents = 1;
+        final int fromSequenceNumber = 10;
+
+        when(managementNode.peek(fromSequenceNumber))
+            .thenReturn(Mono.just(mock(ServiceBusReceivedMessage.class)));
```

Comment:
When you return inline mocks like this, you need to add the Mockito.clearInlinMocks() or else it won't be garbage collected.